### PR TITLE
Mldb 1597 r2

### PR DESCRIPTION
**Pull request by mailletf**
Fixes r2 score calculation. Prevents us from falling into edge cases with previous method

### CODE REVIEW

File: plugins/accuracy.cc [hunk 1]
```diff
@@ -479,29 +480,47 @@ runRegression(AccuracyConfig & runAccuracyConf,
     }
 
 
-
-    double sum_vsq = 0.0, sum_v = 0.0, sum_lsq = 0.0, sum_l = 0.0;
-    double sum_vl = 0.0, n = 0, mse_sum = 0;
-
+    double n = 0, mse_sum = 0;
+    vector<vector<std::array<double, 2>>> allThreadValues;
```
Comment (steadtler) [inline plugins/accuracy.cc]: il'd put spaces to avoid problems on less-than-perfect compilers with the >>

File: plugins/accuracy.cc [hunk 2]
```diff
@@ -479,29 +480,47 @@ runRegression(AccuracyConfig & runAccuracyConf,
...
+                        allThreadValues.emplace_back(std::move(thrStats->values));
                   });
 
 
-    double svl = n * sum_vl - sum_v * sum_l;
-    double svv = n * sum_vsq - sum_v * sum_v;
-    double sll = n * sum_lsq - sum_l * sum_l;
+    std::mutex mergeAccumsLock;
+
+    double meanOfLabel = 0;
```
Comment (steadtler) [inline plugins/accuracy.cc]: Any possibility of zero for n or totalSumSquares

**Comment by steadtler** [comment]
Failing tests

**Comment by nicolaskruchten** [comment]
+1

**Pull request merged**